espnet/nets/batch_beam_search.py

```diff
@@ -183,7 +183,7 @@ def score_full(
         states = dict()
         for k, d in self.full_scorers.items():
             if "decoder" in k and self.return_hs:
-                scores[k], hs, states[k] = d.batch_score(
+                (scores[k], hs), states[k] = d.batch_score(
```

Comment:
I would like to change to this style, however, at the same time, please be aware that the same decoding is used in st task as well (better to  change those together)

Reply:
If I understand you correctly, by changing to style you mean following the format `Tuple[Tuple[xx, xx], xx]`, which is currently the case for `batch_score` and `forward` methods?

Then let me double check references to these methods and make changes accordingly.

Reply:
Another idea I've been thinking about for a while is replacing the return type with a dataclass, which may help future modification to it. This technique is used in huggingface library, and can be potentially apply to many class methods in ESPnet.

Reply:
Sounds good. I like the idea, but I feel since this PR is already large, we might first start the discussion in an issue and then add the changes from another PR though

Reply:
I agree, let me keep the current format and work on this later.